mclagsyncd/mclaglink.cpp

```diff
@@ -0,0 +1,814 @@
+                                fdb.port_name.c_str(), fdb.mac.c_str(), fdb.vid, fdb.type.c_str());
+            }
+            else
+            {
+                SWSS_LOG_NOTICE("modify node(mac =%s, vid =%d, portname :%s ==> %s, type:%s ==>%s)",
+                                fdb.mac.c_str(), fdb.vid, it->port_name.c_str(), fdb.port_name.c_str(), it->type.c_str(), fdb.type.c_str());
+                p_old_fdb->erase(it);
+                p_old_fdb->insert(fdb);
+                #if 0
+                fdb_entry = &(*it);
+                fdb_entry->port_name = fdb.port_name;
+                fdb_entry->type = fdb.type;
+                #endif
+            }
+
+            p_fdb_tbl->set(fdb_key, attrs);
+            SWSS_LOG_NOTICE("add fdb entry into ASIC_DB:key =%s, type =%s", fdb_key.c_str(),  fdb.type.c_str());
+        }
+        else if (fdb_info->op_type == MCLAG_FDB_OPER_DEL)
+        {
+            if (exist)
+            {
+                p_old_fdb->erase(it);
+                SWSS_LOG_NOTICE("erase node(portname =%s, mac =%s, vid =%d, type =%s) from old_fdb_set",
+                                it->port_name.c_str(), it->mac.c_str(), it->vid, it->type.c_str());
+            }
+            p_fdb_tbl->del(fdb_key);
+            SWSS_LOG_NOTICE("del fdb entry from ASIC_DB:key =%s", fdb_key.c_str());
+        }
+    }
+
+    return;
+}
+
+ssize_t  MclagLink::mclagsyncd_get_fdb_changes(char *msg_buf)
+{
+    set <mclag_fdb> new_fdb;
+    set <mclag_fdb> del_fdb;
+    set <mclag_fdb> add_fdb;
+    struct mclag_fdb_info info;
+    mclag_msg_hdr_t * msg_head = NULL;
+    ssize_t write = 0;
+    size_t infor_len = 0;
+    char *infor_start = msg_buf;
+    set <mclag_fdb> *p_new_fdb = &new_fdb;
+
+    del_fdb.clear();
+    add_fdb.clear();
+    p_new_fdb->clear();
+
+    infor_len = infor_len + sizeof(mclag_msg_hdr_t);
+
+    mclagsyncd_get_fdb_set(p_new_fdb);
+
+    set_difference(p_old_fdb->begin(), p_old_fdb->end(), p_new_fdb->begin(),
+                   p_new_fdb->end(), inserter(del_fdb, del_fdb.begin()));
+    set_difference(p_new_fdb->begin(), p_new_fdb->end(), p_old_fdb->begin(),
+                   p_old_fdb->end(), inserter(add_fdb, add_fdb.begin()));
+
+    p_old_fdb->swap(*p_new_fdb);
+
+    for (auto it = del_fdb.begin(); it != del_fdb.end(); it++)
+    {
+        if (MCLAG_MAX_SEND_MSG_LEN - infor_len < sizeof(struct mclag_fdb_info))
+        {
+            msg_head = (mclag_msg_hdr_t *)infor_start;
+            msg_head->version = 1;
+            msg_head->msg_len = (unsigned short)infor_len;
+            msg_head->msg_type = MCLAG_SYNCD_MSG_TYPE_FDB_OPERATION;
+
+            SWSS_LOG_NOTICE("mclagsycnd send msg to iccpd, msg_len =%d, msg_type =%d",
+                            msg_head->msg_len, msg_head->msg_type);
+            write = ::write(m_connection_socket, infor_start, msg_head->msg_len);
+            if (write <= 0)
+                return write;
+
+            infor_len = 0;
```

Comment:
i believe infor_len should be initialised to header size. infor_len=sizeof(mclag_msg_hdr_t); same issue in line 580 also

Reply:
@gitsabari, We will modify it ASAP. Thanks.